.github/workflows/ci.yml

```diff
@@ -80,3 +80,51 @@ jobs:
           tag="${GITHUB_REF#refs/tags/}"
           args=$(printf -- "-a %q " {xtract-core,testing,macros}/target/*/*.jar)
           hub release create $args -m "Release $tag" $tag
+  publish-docs:
+    if: startsWith(github.ref, 'refs/tags')
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          path: code
+      - name: Checkout pages
+        uses: actions/checkout@v2
+        with:
+          ref: refs/heads/gh-pages
+          path: pages
+          fetch-depth: 0
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.ivy2/cache
+            ~/.sbt
+            ~/.coursier
+            target/
+            */target/
+          key: ${{ runner.os }}-sbt-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('**/*.scala') }}
+          restore-keys: |
+            ${{ runner.os }}-sbt-${{ hashFiles('**/*.sbt') }}-
+            ${{ runner.os }}-sbt-
```

Comment:
I feel like I'm missing something with how restore-keys work; from the docs, it sounds like they'll provide a hit if there's any entries in the cache that match this prefix. But that seems like it'd give you false hits? If the main key was missed, that means the Scala source changed. Am I missing something?

TBH this is probably not built enough for a complex cache to be worthwhile. I'd remove the restore-keys form the cache. And in retrospect, I'd remove them from the other jobs' caches too unless you're confident they won't ever cause false hits (and if so, please explain this to me :smile:)

Reply:
The idea here is that even if the source files are different we can still use the cache for upstream packages.  

You're right though, this won't be built very often. I can remove it.